Common/Util/LeanData.cs

```diff
@@ -60,17 +60,48 @@ public static string GenerateLine(IBaseData data, SecurityType securityType, Res
                     {
                         case Resolution.Tick:
                             var tick = (Tick) data;
-                            return ToCsv(milliseconds, Scale(tick.LastPrice), tick.Quantity, tick.Exchange, tick.SaleCondition, tick.Suspicious ? "1" : "0");
-
+                            if (tick.TickType == TickType.Trade)
+                            {
+                                return ToCsv(milliseconds, Scale(tick.LastPrice), tick.Quantity, tick.Exchange, tick.SaleCondition, tick.Suspicious ? "1" : "0");
+                            }
+                            if (tick.TickType == TickType.Quote)
+                            {
+                                return ToCsv(milliseconds, Scale(tick.BidPrice), tick.BidSize, Scale(tick.AskPrice), tick.AskSize, tick.Exchange, tick.SaleCondition, tick.Suspicious ? "1" : "0");
+                            }
+                            break;
                         case Resolution.Minute:
                         case Resolution.Second:
-                            var bar = (TradeBar) data;
-                            return ToCsv(milliseconds, Scale(bar.Open), Scale(bar.High), Scale(bar.Low), Scale(bar.Close), bar.Volume);
+                            var tradeBar = data as TradeBar;
+                            if (tradeBar != null)
+                            {
+                                return ToCsv(milliseconds, Scale(tradeBar.Open), Scale(tradeBar.High), Scale(tradeBar.Low), Scale(tradeBar.Close), tradeBar.Volume);
+                            }
+                            var quoteBar = data as QuoteBar;
+                            if (quoteBar != null)
+                            {
+                                var bidClose = quoteBar.Bid != null ? Scale(quoteBar.Bid.Close) : 0;
+                                var askClose = quoteBar.Ask != null ? Scale(quoteBar.Ask.Close) : 0;
+                                return ToCsv(milliseconds,
+                                    bidClose, quoteBar.LastBidSize,
+                                    askClose, quoteBar.LastAskSize);
```

Comment:
Why are we only using `bigQuoteBar.Bid.Close` and `bidQuoteBar.Ask.Close` here? Shouldn't we be returning all fields so that they can be written to disk here? Maybe I'm not understanding the shape of the data since I'm thinking of bid OHLCV and ask OHLCV bars

Reply:
Good catch!